### DISCUSSION TITLE
How to access generated request ID in request or response object?

### DISCUSSION THREAD

**Discussion by SiNONiMiTY** [fastify/fastify#4813] [General] [answered]
In Fastify logging, you can provide a custom function to genReqId that generates the IDs.

I want to be able to access this value via the request/response object and return it along with the response body. Is this possible?

The reason is that, I want the request log to be easily searchable when the users report an issue.

**Comment by Uzlopak** [comment, marked as answer]
Theoretically you could add a hook in onSend and take the id from request.id and put it into e.g. response.header['x-request-id']

**Reply by SiNONiMiTY** [reply]
Thanks @Uzlopak , I just also found out on the docs that there is an ID property on the FastifyRequest object. Now I can implement the feature that I want to do. :)
